airflow/jobs.py

```diff
@@ -2247,6 +2247,7 @@ def _per_task_process(task, key, ti, session=None):
                             self.log.debug('Sending %s to executor', ti)
                             # Skip scheduled state, we are executing immediately
                             ti.state = State.QUEUED
+                            ti.pool = pool_name
```

Comment:
will pool_name be initialized correctly?

Reply:
yes it is initialized in:
https://github.com/apache/airflow/blob/d9adf1b4a53b07701572f27e2293f23dc8560a10/airflow/jobs.py#L2304

Reply:
That line is _after_ this one.